src/tests/Microsoft.PowerFx.Core.Tests/TypeSystemTests/DTypeTests.cs

```diff
@@ -1144,6 +1145,31 @@ public void TestDefaultSchemaDifference(bool usePowerFxV1CompatibilityRules)
             Assert.Equal(testSchemaDifference.Value, DType.Invalid);
         }
 
+        [Fact]
+        public void OptionSetEquality()
+        {
+            var mutable1 = new Dictionary<DName, DName>() 
+            { 
+                { new DName("A"), new DName("1") },
+                { new DName("B"), new DName("2") } 
+            };
+
+            var immutable = mutable1.ToImmutableDictionary();
+
+            OptionSet optionSet1 = new OptionSet(
+                "option",
+                immutable);
+
+            OptionSet optionSet2 = new OptionSet(
+                "option",
+                immutable);
+
+            var ov1 = new OptionSetValueType(optionSet1);
```

Comment:
I'd add assert that these references are not equal - just to drive home the point and coument the test.